inst/include/cpp11/protect.hpp

```diff
@@ -179,42 +173,50 @@ void unwind_protect(Fun code) {
 struct protect {
   template <typename F>
   struct function;
-  template <typename R, typename... A>
-  struct function<R(A...)> {
-    constexpr R operator()(A... a) const {
+
+  template <typename... A>
+  struct function<void(A...)> {
+    constexpr void operator()(A... a) const {
```

Comment:
At least with `-std=c++11` I am getting `Constexpr function's return type 'void' is not a literal type` for this. I think void constexpr are a C++14 feature.